UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/FileShareService.cs

```diff
@@ -72,16 +73,53 @@ public async Task<Stream> FSSDownloadFileAsync(string batchId, string fileName,
+        }
+
+        public async Task<Stream> FSSDownloadZipFile(string batchId, string fileName, string accessToken, string correlationId)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.FSSGetDailyZipNMFileStarted.ToEventId(), "Maritime safety information request for FSS to get daily zip NM file started for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", batchId, fileName, correlationId);
+
+                FileShareApiClient fileShareApi = new(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+                IResult<Stream> stream = await fileShareApi.DownloadZipFileAsync(batchId, CancellationToken.None);
+                if (stream.IsSuccess)
+                {
+                    _logger.LogInformation(EventIds.FSSGetDailyZipNMFileCompleted.ToEventId(), "Maritime safety information request for FSS to get daily zip NM file completed for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", batchId, fileName, correlationId);
+                    return stream.Data;
+                }
+                else
+                {
+                    _logger.LogInformation(EventIds.FSSGetDailyZipNMFileReturnIsSuccessFalse.ToEventId(), "Maritime safety information request for FSS to get daily zip NM file returns IsSuccess false with StatusCode {StatusCode} for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", stream.StatusCode, batchId, fileName, correlationId);
+
+                    if (stream.Errors.Count > 0)
+                    {
+                        StringBuilder error = new();
+                        foreach (Error item in stream.Errors)
+                            error.AppendLine(item.Description);
+
+                        _logger.LogInformation(EventIds.FSSDownloadZipFileAsyncHasError.ToEventId(), "Maritime safety information request for FSS to get daily zip NM file has error for batchId:{batchId} and fileName:{fileName} with error:{error} for _X-Correlation-ID:{correlationId}", batchId, fileName, error, correlationId);
+                        throw new AggregateException(error.ToString());
+                    }
+                }
             }
             catch (Exception ex)
             {
-                _logger.LogError(EventIds.FSSGetSingleWeeklyNMFileResponseFailed.ToEventId(), "Failed to get single weekly NM file from FSS for batchId:{batchId} and fileName:{fileName} with exception:{exceptionMessage} for _X-Correlation-ID:{CorrelationId}",batchId,fileName, ex.Message, correlationId);
+                _logger.LogError(EventIds.FSSGetDailyZipNMFileResponseFailed.ToEventId(), "Failed to get daily zip NM file from FSS for batchId:{batchId} and fileName:{fileName} with exception:{exceptionMessage} for _X-Correlation-ID:{CorrelationId}", batchId, ex.Message, fileName, correlationId);
                 throw;
             }
+            return null;
```

Comment:
Not covered by any unit tests.

Reply:
We are creating new object of FileShareApiClient inside method. So its always returns stream.IsSuccess as true, Hence code is not testable through unit test

Reply:
Added new test case WhenFSSDownloadZipFileAsyncIsCalled_ThenShouldReturnNull to cover null result